GCC 7 reports various -Wimplicit-fallthrough= errors

GCC properly reports various places in code, where a case label falls through to another label:

json_object.c:678:6: warning: this statement may fall through [-Wimplicit-fallthrough=]
json_tokener.c:339:12: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:418:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:419:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:420:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:421:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:422:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:423:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:424:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:425:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:426:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:427:15: warning: this statement may fall through [-Wimplicit-fallthrough=]
linkhash.c:428:15: warning: this statement may fall through [-Wimplicit-fallthrough=]

Can you please take a look?

--- a/json_object.c
+++ b/json_object.c
@@ -677,6 +677,7 @@ int64_t json_object_get_int64(const struct json_object *jso)
 	case json_type_string:
 		if (json_parse_int64(get_string_component(jso), &cint) == 0)
 			return cint;
+		/* FALLTHRU */
 	default:
 		return 0;
 	}

--- a/json_tokener.c
+++ b/json_tokener.c
@@ -341,6 +341,7 @@ struct json_object* json_tokener_parse_ex(struct json_tokener *tok,
             tok->err = json_tokener_error_parse_unexpected;
             goto out;
         }
+	/* FALLTHRU */
       case '"':
 	state = json_tokener_state_string;
 	printbuf_reset(tok->pb);

--- a/linkhash.c
+++ b/linkhash.c
@@ -415,17 +415,17 @@ static uint32_t hashlittle( const void *key, size_t length, uint32_t initval)
     /*-------------------------------- last block: affect all 32 bits of (c) */
     switch(length)                   /* all the case statements fall through */
     {
-    case 12: c+=((uint32_t)k[11])<<24;
-    case 11: c+=((uint32_t)k[10])<<16;
-    case 10: c+=((uint32_t)k[9])<<8;
-    case 9 : c+=k[8];
-    case 8 : b+=((uint32_t)k[7])<<24;
-    case 7 : b+=((uint32_t)k[6])<<16;
-    case 6 : b+=((uint32_t)k[5])<<8;
-    case 5 : b+=k[4];
-    case 4 : a+=((uint32_t)k[3])<<24;
-    case 3 : a+=((uint32_t)k[2])<<16;
-    case 2 : a+=((uint32_t)k[1])<<8;
+    case 12: c+=((uint32_t)k[11])<<24; /* FALLTHRU */
+    case 11: c+=((uint32_t)k[10])<<16; /* FALLTHRU */
+    case 10: c+=((uint32_t)k[9])<<8; /* FALLTHRU */
+    case 9 : c+=k[8]; /* FALLTHRU */
+    case 8 : b+=((uint32_t)k[7])<<24; /* FALLTHRU */
+    case 7 : b+=((uint32_t)k[6])<<16; /* FALLTHRU */
+    case 6 : b+=((uint32_t)k[5])<<8; /* FALLTHRU */
+    case 5 : b+=k[4]; /* FALLTHRU */
+    case 4 : a+=((uint32_t)k[3])<<24; /* FALLTHRU */
+    case 3 : a+=((uint32_t)k[2])<<16; /* FALLTHRU */
+    case 2 : a+=((uint32_t)k[1])<<8; /* FALLTHRU */
     case 1 : a+=k[0];
              break;
     case 0 : return c;

